### PR TITLE
HARVESTER: VM template is not working with Node scheduling

### PR DESCRIPTION
**Pull request by n313893254**
### Summary
Fixes VM template is not working with Node scheduling

### Occurred changes and/or fixed issues

- https://github.com/harvester/harvester/issues/2244

### Technical notes summary

- Add node scheduling for template
- Add validate for node scheduling node name

### Screenshot/Video

image

image

### CODE REVIEW

**Review by DaiYuzeng — APPROVED** [review]
LGTM!

File: shell/components/form/NodeScheduling.vue [hunk 1]
```diff
@@ -127,7 +127,33 @@ export default {
...
+          if (nodeName && !array.includes(nodeName)) {
+            this.$store.dispatch('growl/error', {
+              title:   this.$store.getters['i18n/t']('harvester.vmTemplate.tips.notExistNode.title', { name: nodeName }),
+              message: this.$store.getters['i18n/t']('harvester.vmTemplate.tips.notExistNode.message')
+            }, { root: true });
+          }
+
+          delete this.value.nodeSelector;
+          this.$set(this, 'nodeName', '');
+          this.$set(this, 'selectNode', null);
```
Comment (WuJun2016) [inline shell/components/form/NodeScheduling.vue:152]: why delete?

Data is cleared when saved.

File: shell/components/form/NodeScheduling.vue [hunk 2]
```diff
@@ -127,7 +127,33 @@ export default {
...
+
+          if (nodeName && !array.includes(nodeName)) {
+            this.$store.dispatch('growl/error', {
+              title:   this.$store.getters['i18n/t']('harvester.vmTemplate.tips.notExistNode.title', { name: nodeName }),
+              message: this.$store.getters['i18n/t']('harvester.vmTemplate.tips.notExistNode.message')
+            }, { root: true });
+          }
+
+          delete this.value.nodeSelector;
+          this.$set(this, 'nodeName', '');
```
Comment (WuJun2016) [inline shell/components/form/NodeScheduling.vue:151]: I wonder if we could move this section to mixins/harvester-vm, which would affect the rancher's logic.

File: shell/components/nav/Header.vue [hunk 3]
```diff
@@ -329,7 +329,7 @@ export default {
     <div class="spacer"></div>
 
     <div class="rd-header-right">
-      <HarvesterUpgrade v-if="isVirtualCluster" />
+      <HarvesterUpgrade v-if="isVirtualCluster && currentProduct.name === 'harvester'" />
```
Comment (WuJun2016) [inline shell/components/nav/Header.vue:332]: use   
import { NAME as HARVESTER } from '@shell/config/product/harvester';

currentProduct.name === HARVESTER

**Review by WuJun2016 — CHANGES_REQUESTED** [review]

File: shell/components/form/NodeScheduling.vue (hunk 2 above)
Comment (n313893254) [reply shell/components/form/NodeScheduling.vue:151]: I am afraid not, need to set nodeName and selectNode which belongs to the component to empty.

File: shell/components/form/NodeScheduling.vue (hunk 1 above)
Comment (n313893254) [reply shell/components/form/NodeScheduling.vue:152]: It also needs to empty when viewing the detail page. same as https://github.com/harvester/dashboard/blob/a0e953e9348d2ba271184428de605a08db5077e1/shell/edit/kubevirt.io.virtualmachine/VirtualMachineVolume/type/vmImage.vue#L171

**Review by WuJun2016 — APPROVED** [review]
LGTM

**Pull request merged**